### ISSUE TITLE
weird error when adding websocket-next to extension project

### ISSUE THREAD

**Issue by maxandersen** [quarkusio/quarkus#42056]
### Describe the bug

I have a working project using old websockets; now trying to move to using websockets-next (just client) in an extension.

mainly opening this to have somewhere on the internet this stacktrace to help people may be in same situation.

Only thing i  did thus far as adding websockets-next to the runtime and deployment project and I get this when running:

```
[quarkus-homeassistant-deployment] [ERROR] io.quarkiverse.homeassistant.test.HomeAssistantDevModeTest.writeYourOwnDevModeTest -- Time elapsed: 3.773 s <<< ERROR!
java.lang.RuntimeException: 
io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.arc.deployment.ArcProcessor#validate threw an exception: jakarta.enterprise.inject.spi.DeploymentException: jakarta.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type io.quarkus.websockets.next.runtime.ConnectionManager and qualifiers [@Default]
        - injection target: parameter 'connectionManager' of io.quarkus.websockets.next.runtime.devui.WebSocketNextJsonRPCService constructor
        - declared on CLASS bean [types=[io.quarkus.websockets.next.runtime.ConnectionManager$ConnectionListener, io.quarkus.websockets.next.runtime.devui.WebSocketNextJsonRPCService, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.websockets.next.runtime.devui.WebSocketNextJsonRPCService]
        at io.quarkus.arc.processor.BeanDeployment.processErrors(BeanDeployment.java:1546)
        at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:338)
        at io.quarkus.arc.processor.BeanProcessor.initialize(BeanProcessor.java:167)
        at io.quarkus.arc.deployment.ArcProcessor.validate(ArcProcessor.java:490)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:849)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2516)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2495)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1521)
        at java.base/java.lang.Thread.run(Thread.java:1583)
        at org.jboss.threads.JBossThread.run(JBossThread.java:483)
Caused by: jakarta.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type io.quarkus.websockets.next.runtime.ConnectionManager and qualifiers [@Default]
        - injection target: parameter 'connectionManager' of io.quarkus.websockets.next.runtime.devui.WebSocketNextJsonRPCService constructor
        - declared on CLASS bean [types=[io.quarkus.websockets.next.runtime.ConnectionManager$ConnectionListener, io.quarkus.websockets.next.runtime.devui.WebSocketNextJsonRPCService, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.websockets.next.runtime.devui.WebSocketNextJsonRPCService]
        at io.quarkus.arc.processor.Beans.resolveInjectionPoint(Beans.java:519)
        at io.quarkus.arc.processor.BeanInfo.init(BeanInfo.java:642)
        at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:323)
        ... 12 more

        at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:331)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:252)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:58)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:113)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:435)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:56)
        at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:138)
        at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:93)
        at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:131)
        at io.quarkus.test.QuarkusDevModeTest.beforeEach(QuarkusDevModeTest.java:304)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.arc.deployment.ArcProcessor#validate threw an exception: jakarta.enterprise.inject.spi.DeploymentException: jakarta.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type io.quarkus.websockets.next.runtime.ConnectionManager and qualifiers [@Default]
        - injection target: parameter 'connectionManager' of io.quarkus.websockets.next.runtime.devui.WebSocketNextJsonRPCService constructor
        - declared on CLASS bean [types=[io.quarkus.websockets.next.runtime.ConnectionManager$ConnectionListener, io.quarkus.websockets.next.runtime.devui.WebSocketNextJsonRPCService, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.websockets.next.runtime.devui.WebSocketNextJsonRPCService]
        at io.quarkus.arc.processor.BeanDeployment.processErrors(BeanDeployment.java:1546)
        at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:338)
        at io.quarkus.arc.processor.BeanProcessor.initialize(BeanProcessor.java:167)
        at io.quarkus.arc.deployment.ArcProcessor.validate(ArcProcessor.java:490)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:849)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2516)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2495)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1521)
        at java.base/java.lang.Thread.run(Thread.java:1583)
        at org.jboss.threads.JBossThread.run(JBossThread.java:483)
Caused by: jakarta.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type io.quarkus.websockets.next.runtime.ConnectionManager and qualifiers [@Default]
        - injection target: parameter 'connectionManager' of io.quarkus.websockets.next.runtime.devui.WebSocketNextJsonRPCService constructor
        - declared on CLASS bean [types=[io.quarkus.websockets.next.runtime.ConnectionManager$ConnectionListener, io.quarkus.websockets.next.runtime.devui.WebSocketNextJsonRPCService, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.websockets.next.runtime.devui.WebSocketNextJsonRPCService]
        at io.quarkus.arc.processor.Beans.resolveInjectionPoint(Beans.java:519)
        at io.quarkus.arc.processor.BeanInfo.init(BeanInfo.java:642)
        at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:323)
        ... 12 more

        at io.quarkus.builder.Execution.run(Execution.java:123)
        at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:79)
        at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:160)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:327)
        ... 11 more
Caused by: jakarta.enterprise.inject.spi.DeploymentException: jakarta.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type io.quarkus.websockets.next.runtime.ConnectionManager and qualifiers [@Default]
        - injection target: parameter 'connectionManager' of io.quarkus.websockets.next.runtime.devui.WebSocketNextJsonRPCService constructor
        - declared on CLASS bean [types=[io.quarkus.websockets.next.runtime.ConnectionManager$ConnectionListener, io.quarkus.websockets.next.runtime.devui.WebSocketNextJsonRPCService, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.websockets.next.runtime.devui.WebSocketNextJsonRPCService]
        at io.quarkus.arc.processor.BeanDeployment.processErrors(BeanDeployment.java:1546)
        at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:338)
        at io.quarkus.arc.processor.BeanProcessor.initialize(BeanProcessor.java:167)
        at io.quarkus.arc.deployment.ArcProcessor.validate(ArcProcessor.java:490)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:849)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
        at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
        at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2516)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2495)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1521)
        at java.base/java.lang.Thread.run(Thread.java:1583)
        at org.jboss.threads.JBossThread.run(JBossThread.java:483)
Caused by: jakarta.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type io.quarkus.websockets.next.runtime.ConnectionManager and qualifiers [@Default]
        - injection target: parameter 'connectionManager' of io.quarkus.websockets.next.runtime.devui.WebSocketNextJsonRPCService constructor
        - declared on CLASS bean [types=[io.quarkus.websockets.next.runtime.ConnectionManager$ConnectionListener, io.quarkus.websockets.next.runtime.devui.WebSocketNextJsonRPCService, java.lang.Object], qualifiers=[@Default, @Any], target=io.quarkus.websockets.next.runtime.devui.WebSocketNextJsonRPCService]
        at io.quarkus.arc.processor.Beans.resolveInjectionPoint(Beans.java:519)
        at io.quarkus.arc.processor.BeanInfo.init(BeanInfo.java:642)
        at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:323)
        ... 11 more
```

### Expected behavior

no errors just adding exception - or if something that can happen some better error message.

### How to Reproduce?

```
git clone https://github.com/maxandersen/quarkus-homeassistant.git
git checkout brokennext
mvn clean package
```

code: https://github.com/maxandersen/quarkus-homeassistant/tree/brokennext

### Quarkus version or git rev

g

### Additional information

related to this, which dependency/config needed if only doing websocket client not serveR?

**Comment by geoand** [comment]
cc @mkouba

**Comment by mkouba** [comment]
Hey Max, yes, this problems occurs in the dev mode (it's actually a Dev UI issue) when no server endpoint is defined and should be fixed in 3.12.2+ and 3.13+. It was fixed in  https://github.com/quarkusio/quarkus/pull/41779 and discussed [on zulip](https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/WebSocket-next.20Client.20question).
